File: app/src/main/java/com/tstudioz/fax/fme/fragments/Prisutnost.java
package com.tstudioz.fax.fme.fragments;


import android.content.Context;
import android.content.pm.ActivityInfo;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.widget.NestedScrollView;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import com.franmontiel.persistentcookiejar.PersistentCookieJar;
import com.franmontiel.persistentcookiejar.cache.SetCookieCache;
import com.franmontiel.persistentcookiejar.persistence.SharedPrefsCookiePersistor;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.tstudioz.fax.fme.R;
import com.tstudioz.fax.fme.adapters.DolasciAdapter;
import com.tstudioz.fax.fme.database.Dolazak;
import com.tstudioz.fax.fme.database.Korisnik;
import com.tstudioz.fax.fme.migrations.CredMigration;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.IOException;
import java.util.StringTokenizer;
import java.util.UUID;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.realm.Realm;
import io.realm.RealmConfiguration;
import io.realm.RealmResults;
import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.CookieJar;
import okhttp3.FormBody;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;


public class Prisutnost extends Fragment {

    public RealmConfiguration realmConfig = new RealmConfiguration.Builder()
            .name("prisutnost.realm")
            .schemaVersion(10)
            .deleteRealmIfMigrationNeeded()
            .build();

    public RealmConfiguration CredRealmCf = new RealmConfiguration.Builder()
            .name("encrypted.realm")
            .schemaVersion(6)
            .migration(new CredMigration())
            .build();

    @BindView(R.id.recyclerZimski) RecyclerView zRecyclerview;
    @BindView(R.id.recyclerLItnji) RecyclerView lRecyclerview;
    @BindView(R.id.progress_attend) ProgressBar mProgress;
    @BindView(R.id.nested_attend) NestedScrollView mNested;

    private Snackbar snack;
    private DolasciAdapter winterAdapter, summerAdapter;
    private Realm nRealm, cRealm, sRealm, wRealm;
    private OkHttpClient okHttpClient;

    private AdView mAdView;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        setHasOptionsMenu(true);
        View view = inflater.inflate(R.layout.prisutnost_tab, container, false);

        ButterKnife.bind(this, view);
        getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        mNested.setVisibility(View.INVISIBLE);
        mProgress.setVisibility(View.VISIBLE);

        mAdView = view.findViewById(R.id.adView);

        startFetching();
        loadAds();

        return view;
    }

    @Override
    public void onResume() {
        super.onResume();

    }

  public void fetchPrisutnost(){

            final CookieJar cookieJar = new PersistentCookieJar(new SetCookieCache(), new SharedPrefsCookiePersistor(getActivity()));

            okHttpClient = new OkHttpClient().newBuilder()
                    .followRedirects(true)
                    .followSslRedirects(true)
                    .cookieJar(cookieJar)
                    .build();

                    nRealm = Realm.getInstance(realmConfig);
                    final RealmResults<Dolazak> svaPrisutnost = nRealm.where(Dolazak.class).findAll();

                    nRealm.executeTransaction(new Realm.Transaction() {
                        @Override
                        public void execute(Realm realm) {
                            svaPrisutnost.deleteAllFromRealm();
                        }
                    });

                    if(getActivity()!=null) {
                        getActivity().runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                showRecyclerviewWinterSem();
                                showRecyclerviewSummerSem();
                            }
                        });
                    }

                    cRealm = Realm.getInstance(CredRealmCf);
                    Korisnik korisnik = cRealm.where(Korisnik.class).findFirst();

                    final RequestBody formData = new FormBody.Builder()
                            .add("Username", korisnik.getUsername())
                            .add("Password", korisnik.getLozinka())
                            .add("IsRememberMeChecked", "true")
                            .build();

                    final Request rq = new Request.Builder()
                            .url("https://korisnik.fesb.unist.hr/prijava?returnUrl=https://raspored.fesb.unist.hr")
                            .post(formData)
                            .build();

                    Call call0 = okHttpClient.newCall(rq);
                    call0.enqueue(new Callback() {
                        @Override
                        public void onFailure(Call call, IOException e) {
                            Log.d("pogreska", "failure");
                        }

                        @Override
                        public void onResponse(Call call, Response response) throws IOException {

                            Request request = new Request.Builder()
                                    .url("https://raspored.fesb.unist.hr/part/prisutnost/opcenito/tablica")
                                    .get()
                                    .build();

                            Call call1 = okHttpClient.newCall(request);
                            call1.enqueue(new Callback() {
                                @Override
                                public void onFailure(Call call, IOException e) {
                                    Log.d("pogreska", "failure");
                                }

                                @Override
                                public void onResponse(Call call, Response response) throws IOException {

                                    if (response.code() != 500) {

                                        Document doc = Jsoup.parse(response.body().string());

                                        try {
                                            Element zimski = doc.select("div.semster.winter").first();
                                            Element litnji = doc.select("div.semster.summer").first();
                                            Element zimskaPredavanja = zimski.select("div.body.clearfix").first();
                                            Element litnjaPredavanja = litnji.select("div.body.clearfix").first();


                                            if (zimski.getElementsByClass("emptyList").first() == null) {

                                                Elements zimskiKolegiji = zimskaPredavanja.select("a");

                                                for (final Element element : zimskiKolegiji) {

                                                    Request request = new Request.Builder()
                                                            .url("https://raspored.fesb.unist.hr" + element.attr("href").toString())
                                                            .get()
                                                            .build();

                                                    Call callonme = okHttpClient.newCall(request);
                                                    callonme.enqueue(new Callback() {
                                                        @Override
                                                        public void onFailure(Call call, IOException e) {
                                                            Log.d("pogreska", "failure");
                                                        }

                                                        @Override
                                                        public void onResponse(Call call, Response response) throws IOException {
                                                            Document document = Jsoup.parse(response.body().string());
                                                            final Realm mRealm1 = Realm.getInstance(realmConfig);

                                                            try {

                                                                Element content = document.getElementsByClass("courseCategories").first();
                                                                final Elements kategorije = content.select("div.courseCategory");

                                                                mRealm1.executeTransaction(new Realm.Transaction() {
                                                                    @Override
                                                                    public void execute(Realm realm) {
                                                                        for (Element kat : kategorije) {
                                                                            final Dolazak mDolazak = mRealm1.createObject(Dolazak.class, UUID.randomUUID().toString());

                                                                            mDolazak.setSemestar(1);
                                                                            mDolazak.setPredmet(element.select("div.cellContent").first().text());
                                                                            mDolazak.setVrsta(kat.getElementsByClass("name").first().text());
                                                                            mDolazak.setAttended(Integer.parseInt(kat.select("div.attended > span.num").first().text()));
                                                                            mDolazak.setAbsent(Integer.parseInt(kat.select("div.absent > span.num").first().text()));
                                                                            mDolazak.setRequired(kat.select("div.required-attendance > span").first().text());

                                                                            String string = kat.select("div.required-attendance > span").first().text();
                                                                            StringTokenizer st = new StringTokenizer(string, " ");
                                                                            String ric1 = st.nextToken();
                                                                            String ric2 = st.nextToken();
                                                                            String max = st.nextToken();

                                                                            mDolazak.setTotal(Integer.parseInt(max));

                                                                        }
                                                                    }
                                                                });

                                                            } catch (Exception exception) {
                                                                Log.d("Exception prisutnost", exception.getMessage());
                                                                exception.printStackTrace();
                                                            } finally {
                                                                mRealm1.close();
                                                            }

                                                        }

                                                    });


                                                }
                                            }


                                            if (litnji.getElementsByClass("emptyList").first() == null) {

                                                Elements litnjiKolegiji = litnjaPredavanja.select("a");

                                                for (final Element element : litnjiKolegiji) {

                                                    Request request = new Request.Builder()
                                                            .url("https://raspored.fesb.unist.hr" + element.attr("href").toString())
                                                            .get()
                                                            .build();

                                                    Call callonme1 = okHttpClient.newCall(request);
                                                    callonme1.enqueue(new Callback() {
                                                        @Override
                                                        public void onFailure(Call call, IOException e) {
                                                            showSnackError();
                                                        }

                                                        @Override
                                                        public void onResponse(Call call, Response response) throws IOException {
                                                            Document document = Jsoup.parse(response.body().string());

                                                            final Realm mRealm2 = Realm.getInstance(realmConfig);

                                                            try {
                                                                Element content = document.getElementsByClass("courseCategories").first();
                                                                final Elements kategorije = content.select("div.courseCategory");

                                                                mRealm2.executeTransaction(new Realm.Transaction() {
                                                                    @Override
                                                                    public void execute(Realm realm) {
                                                                        for (Element kat : kategorije) {
                                                                            final Dolazak mDolazak = mRealm2.createObject(Dolazak.class, UUID.randomUUID().toString());

                                                                            mDolazak.setSemestar(2);
                                                                            mDolazak.setPredmet(element.select("div.cellContent").first().text());
                                                                            mDolazak.setVrsta(kat.getElementsByClass("name").first().text());
                                                                            mDolazak.setAttended(Integer.parseInt(kat.select("div.attended > span.num").first().text()));
                                                                            mDolazak.setAbsent(Integer.parseInt(kat.select("div.absent > span.num").first().text()));
                                                                            mDolazak.setRequired(kat.select("div.required-attendance > span").first().text());

                                                                            String string = kat.select("div.required-attendance > span").first().text();
                                                                            StringTokenizer st = new StringTokenizer(string, " ");
                                                                            String ric1 = st.nextToken();
                                                                            String ric2 = st.nextToken();
                                                                            String max = st.nextToken();

                                                                            mDolazak.setTotal(Integer.parseInt(max));

                                                                        }
                                                                    }
                                                                });

                                                            } catch (Exception e) {
                                                                Log.d("Exception prisutnost", e.getMessage());
                                                                e.printStackTrace();
                                                            } finally {
                                                                mRealm2.close();
                                                            }
                                                        }
                                                    });

                                                }
                                            }

                                        } catch (Exception ex){
                                            Log.d("Exception pris", ex.getMessage());
                                            ex.printStackTrace();
                                        }

                                        if(getActivity()!=null) {
                                            getActivity().runOnUiThread(new Runnable() {
                                                @Override
                                                public void run() {
                                                    updateAdapters();
                                                    mProgress.setVisibility(View.INVISIBLE);
                                                    mNested.setVisibility(View.VISIBLE);
                                                }
                                            });
                                        }

                                    }else {
                                        showSnackError();
                                    }
                                }

                            });
                        }
                    });

    }

    public void showRecyclerviewWinterSem() {

        wRealm = Realm.getInstance(realmConfig);
        RealmResults<Dolazak> dolasciWinter = wRealm.where(Dolazak.class).equalTo("semestar", 1).findAll();

        zRecyclerview.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false));
        winterAdapter = new DolasciAdapter(dolasciWinter);
        zRecyclerview.setAdapter(winterAdapter);


    }

    public void showRecyclerviewSummerSem() {

        sRealm = Realm.getInstance(realmConfig);
        RealmResults<Dolazak> dolasciSummer = sRealm.where(Dolazak.class).equalTo("semestar", 2).findAll();

        lRecyclerview.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false));
        summerAdapter= new DolasciAdapter(dolasciSummer);
        lRecyclerview.setAdapter(summerAdapter);

    }

    public void updateAdapters(){
        winterAdapter.notifyDataSetChanged();
        summerAdapter.notifyDataSetChanged();
    }

    public void loadAds(){
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);
    }


    @Override
    public void onPrepareOptionsMenu(Menu menu) {
        menu.findItem(R.id.refresMe).setVisible(false);
        super.onPrepareOptionsMenu(menu);
    }

    public void startFetching(){
        if(isNetworkAvailable()) {
            fetchPrisutnost();
        }else {
            showSnacOffline();
        }
    }

    public void showSnacOffline(){
        snack = Snackbar.make(getActivity().findViewById(R.id.coordinatorLayout), "Niste povezani", Snackbar.LENGTH_INDEFINITE);
        View vjuz = snack.getView();
        vjuz.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.red_nice));
        snack.setAction("PONOVI", new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                snack.dismiss();
                startFetching();
            }
        });
        snack.setActionTextColor(getResources().getColor(R.color.white));
        snack.show();
    }

    public void showSnackError(){
        snack=Snackbar.make(getActivity().findViewById(R.id.coordinatorLayout), "Došlo je do pogreške", Snackbar.LENGTH_LONG);
        View okvir = snack.getView();
        okvir.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.red_nice));
        snack.setAction(("PONOVI"), new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                snack.dismiss();
                startFetching();
            }
        });
        snack.setActionTextColor(getResources().getColor(R.color.white));
        snack.show();
    }


    private boolean isNetworkAvailable() {
        ConnectivityManager manager = (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = manager.getActiveNetworkInfo();

        boolean isAvailable = false;
        if (networkInfo != null && networkInfo.isConnected()) {
            isAvailable = true;
        }
        return isAvailable;
    }

    public void onStop(){
        super.onStop();

        if(snack!=null){
            snack.dismiss();
        }

        if(okHttpClient!=null)
            okHttpClient.dispatcher().cancelAll();
    }

    @Override
    public void onDestroy(){
        super.onDestroy();

        if(nRealm!=null)
            nRealm.close();

        if(cRealm!=null)
            cRealm.close();

        if(sRealm!=null)
            sRealm.close();

        if(wRealm!=null)
            wRealm.close();

    }

}

File: app/src/main/java/com/tstudioz/fax/fme/fragments/Prisutnost.kt
package com.tstudioz.fax.fme.fragments

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.Menu
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.franmontiel.persistentcookiejar.PersistentCookieJar
import com.franmontiel.persistentcookiejar.cache.SetCookieCache
import com.franmontiel.persistentcookiejar.persistence.SharedPrefsCookiePersistor
import com.google.android.material.snackbar.Snackbar
import com.tstudioz.fax.fme.Application.FESBCompanion.Companion.instance
import com.tstudioz.fax.fme.R
import com.tstudioz.fax.fme.adapters.DolasciAdapter
import com.tstudioz.fax.fme.database.Dolazak
import com.tstudioz.fax.fme.database.Korisnik
import com.tstudioz.fax.fme.databinding.PrisutnostTabBinding
import com.tstudioz.fax.fme.networking.NetworkUtils
import io.realm.Realm
import io.realm.RealmConfiguration
import io.realm.RealmResults
import kotlinx.coroutines.InternalCoroutinesApi
import okhttp3.Call
import okhttp3.Callback
import okhttp3.CookieJar
import okhttp3.FormBody.Builder
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody
import okhttp3.Response
import org.jsoup.Jsoup
import java.io.IOException
import java.util.StringTokenizer
import java.util.UUID

class Prisutnost : Fragment() {
    var realmConfig: RealmConfiguration = RealmConfiguration.Builder()
        .allowWritesOnUiThread(true)
        .name("prisutnost.realm")
        .schemaVersion(10)
        .deleteRealmIfMigrationNeeded()
        .build()
    private var snack: Snackbar? = null
    private var winterAdapter: DolasciAdapter? = null
    private var summerAdapter: DolasciAdapter? = null
    private var nRealm: Realm? = null
    private var cRealm: Realm? = null
    private var sRealm: Realm? = null
    private var wRealm: Realm? = null
    private var okHttpClient: OkHttpClient? = null
    private var binding: PrisutnostTabBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        setHasOptionsMenu(true)
        binding = PrisutnostTabBinding.inflate(inflater, container, false)
        requireActivity().requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
        hideRecyc()
        startFetching()
        return binding!!.root
    }

    @OptIn(InternalCoroutinesApi::class)
    fun fetchPrisutnost() {
        val cookieJar: CookieJar = PersistentCookieJar(
            SetCookieCache(),
            SharedPrefsCookiePersistor(activity)
        )
        okHttpClient = instance!!.okHttpInstance
        deletePreviousResults()
        cRealm = Realm.getDefaultInstance()
        val korisnik = cRealm?.where(Korisnik::class.java)?.findFirst()
        val formData: RequestBody = Builder()
            .add("Username", korisnik!!.getUsername())
            .add("Password", korisnik.getLozinka())
            .add("IsRememberMeChecked", "true")
            .build()
        val rq: Request = Request.Builder()
            .url(
                "https://korisnik.fesb.unist.hr/prijava?returnUrl=https://raspored.fesb" +
                        ".unist.hr"
            )
            .post(formData)
            .build()
        val call0 = okHttpClient!!.newCall(rq)
        call0.enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {
                Log.d("pogreska", "failure")
            }

            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                val request: Request = Request.Builder()
                    .url("https://raspored.fesb.unist.hr/part/prisutnost/opcenito/tablica")
                    .get()
                    .build()
                val call1 = okHttpClient!!.newCall(request)
                call1.enqueue(object : Callback {
                    override fun onFailure(call: Call, e: IOException) {
                        Log.d("pogreska", "failure")
                    }

                    @Throws(IOException::class)
                    override fun onResponse(call: Call, response: Response) {
                        if (response.code != 500) {
                            val doc = Jsoup.parse(response.body!!.string())
                            try {
                                val zimski = doc.select("div.semster.winter").first()
                                val litnji = doc.select("div.semster.summer").first()
                                val zimskaPredavanja = zimski.select("div.body.clearfix").first()
                                val litnjaPredavanja = litnji.select("div.body.clearfix").first()
                                if (zimski.getElementsByClass("emptyList").first() == null) {
                                    val zimskiKolegiji = zimskaPredavanja.select("a")
                                    for (element in zimskiKolegiji) {
                                        val request: Request = Request.Builder()
                                            .url(
                                                "https://raspored.fesb.unist.hr" + element.attr("href")
                                                    .toString()
                                            )
                                            .get()
                                            .build()
                                        val callonme = okHttpClient!!.newCall(request)
                                        callonme.enqueue(object : Callback {
                                            override fun onFailure(call: Call, e: IOException) {
                                                Log.d("pogreska", "failure")
                                            }

                                            @Throws(IOException::class)
                                            override fun onResponse(
                                                call: Call,
                                                response: Response
                                            ) {
                                                val document = Jsoup.parse(
                                                    response.body!!.string()
                                                )
                                                val mRealm1 = Realm.getInstance(realmConfig)
                                                try {
                                                    val content = document.getElementsByClass(
                                                        "courseCategories"
                                                    ).first()
                                                    val kategorije = content.select(
                                                        "div.courseCategory"
                                                    )
                                                    mRealm1.executeTransaction { realm ->
                                                        for (kat in kategorije) {
                                                            val mDolazak = realm.createObject(
                                                                Dolazak::class.java,
                                                                UUID.randomUUID().toString()
                                                            )
                                                            mDolazak.setSemestar(1)
                                                            mDolazak.setPredmet(
                                                                element.select("div.cellContent")
                                                                    .first().text()
                                                            )
                                                            mDolazak.setVrsta(
                                                                kat.getElementsByClass(
                                                                    "name"
                                                                ).first().text()
                                                            )
                                                            mDolazak.setAttended(
                                                                kat.select("div.attended > span.num")
                                                                    .first().text().toInt()
                                                            )
                                                            mDolazak.setAbsent(
                                                                kat.select("div.absent > span.num")
                                                                    .first().text().toInt()
                                                            )
                                                            mDolazak.setRequired(
                                                                kat.select(
                                                                    "div.required-attendance " +
                                                                            "> span"
                                                                ).first().text()
                                                            )
                                                            val string = kat.select(
                                                                "div" +
                                                                        ".required-attendance > " +
                                                                        "span"
                                                            ).first().text()
                                                            val st = StringTokenizer(string, " ")
                                                            val ric1 = st.nextToken()
                                                            val ric2 = st.nextToken()
                                                            val max = st.nextToken()
                                                            mDolazak.setTotal(max.toInt())
                                                        }
                                                    }
                                                } catch (exception: Exception) {
                                                    Log.d(
                                                        "Exception prisutnost",
                                                        exception.message!!
                                                    )
                                                    exception.printStackTrace()
                                                } finally {
                                                    mRealm1.close()
                                                }
                                            }
                                        })
                                    }
                                }
                                if (litnji.getElementsByClass("emptyList").first() == null) {
                                    val litnjiKolegiji = litnjaPredavanja.select("a")
                                    for (element in litnjiKolegiji) {
                                        val request: Request = Request.Builder()
                                            .url(
                                                "https://raspored.fesb.unist.hr" + element.attr("href")
                                                    .toString()
                                            )
                                            .get()
                                            .build()
                                        val callonme1 = okHttpClient!!.newCall(request)
                                        callonme1.enqueue(object : Callback {
                                            override fun onFailure(call: Call, e: IOException) {
                                                showSnackError()
                                            }

                                            @Throws(IOException::class)
                                            override fun onResponse(
                                                call: Call,
                                                response: Response
                                            ) {
                                                val document = Jsoup.parse(
                                                    response.body!!.string()
                                                )
                                                val mRealm2 = Realm.getInstance(realmConfig)
                                                try {
                                                    val content = document.getElementsByClass(
                                                        "courseCategories"
                                                    ).first()
                                                    val kategorije = content.select(
                                                        "div.courseCategory"
                                                    )
                                                    mRealm2.executeTransaction { realm ->
                                                        for (kat in kategorije) {
                                                            val mDolazak = realm.createObject(
                                                                Dolazak::class.java,
                                                                UUID.randomUUID().toString()
                                                            )
                                                            mDolazak.setSemestar(2)
                                                            mDolazak.setPredmet(
                                                                element.select("div.cellContent")
                                                                    .first().text()
                                                            )
                                                            mDolazak.setVrsta(
                                                                kat.getElementsByClass(
                                                                    "name"
                                                                ).first().text()
                                                            )
                                                            mDolazak.setAttended(
                                                                kat.select("div.attended > span.num")
                                                                    .first().text().toInt()
                                                            )
                                                            mDolazak.setAbsent(
                                                                kat.select("div.absent > span.num")
                                                                    .first().text().toInt()
                                                            )
                                                            mDolazak.setRequired(
                                                                kat.select(
                                                                    "div.required-attendance " +
                                                                            "> span"
                                                                ).first().text()
                                                            )
                                                            val string = kat.select(
                                                                "div" +
                                                                        ".required-attendance > " +
                                                                        "span"
                                                            ).first().text()
                                                            val st = StringTokenizer(string, " ")
                                                            val ric1 = st.nextToken()
                                                            val ric2 = st.nextToken()
                                                            val max = st.nextToken()
                                                            mDolazak.setTotal(max.toInt())
                                                        }
                                                    }
                                                } catch (e: Exception) {
                                                    Log.d("Exception prisutnost", e.message!!)
                                                    e.printStackTrace()
                                                } finally {
                                                    mRealm2.close()
                                                }
                                            }
                                        })
                                    }
                                }
                            } catch (ex: Exception) {
                                Log.d("Exception pris", ex.message!!)
                                ex.printStackTrace()
                            }
                            if (activity != null) {
                                activity!!.runOnUiThread {
                                    showRecyclerviewWinterSem()
                                    showRecyclerviewSummerSem()
                                    binding!!.progressAttend.visibility = View.INVISIBLE
                                    binding!!.nestedAttend.visibility = View.VISIBLE
                                }
                            }
                        } else {
                            showSnackError()
                        }
                    }
                })
            }
        })
    }

    fun showRecyclerviewWinterSem() {
        wRealm = Realm.getInstance(realmConfig)
        val dolasciWinter: RealmResults<Dolazak>? =
            wRealm?.where(Dolazak::class.java)?.equalTo("semestar", 1.toInt())?.findAll()
        try {
            binding!!.recyclerZimski.layoutManager = LinearLayoutManager(activity,
                LinearLayoutManager.HORIZONTAL, false
            )
            if (dolasciWinter!=null)
                winterAdapter = context?.let { DolasciAdapter(it, dolasciWinter) }
            binding!!.recyclerZimski.adapter = winterAdapter
        } finally {
            wRealm?.close()
        }
    }

    fun showRecyclerviewSummerSem() {
        sRealm = Realm.getInstance(realmConfig)
        val dolasciSummer: RealmResults<Dolazak>? =
            sRealm?.where(Dolazak::class.java)?.equalTo("semestar", 2.toInt())?.findAll()
        try {
            binding!!.recyclerLItnji.layoutManager = LinearLayoutManager(activity,
                LinearLayoutManager.HORIZONTAL, false
            )
            if(dolasciSummer!=null)
                summerAdapter = context?.let { DolasciAdapter(it, dolasciSummer) }
            binding!!.recyclerLItnji.adapter = summerAdapter
        } finally {
            sRealm?.close()
        }
    }

    private fun deletePreviousResults() {
        nRealm = Realm.getInstance(realmConfig)
        val svaPrisutnost = nRealm?.where(Dolazak::class.java)?.findAll()
        nRealm?.executeTransaction { svaPrisutnost?.deleteAllFromRealm() }
    }

    override fun onPrepareOptionsMenu(menu: Menu) {
        menu.findItem(R.id.refresMe).isVisible = false
        super.onPrepareOptionsMenu(menu)
    }

    private fun startFetching() {
        if (NetworkUtils.isNetworkAvailable(requireContext())) {
            fetchPrisutnost()
        } else {
            showSnacOffline()
        }
    }

    private fun showSnacOffline() {
        snack = Snackbar.make(
            requireActivity().findViewById(R.id.coordinatorLayout), "Niste " +
                    "povezani", Snackbar.LENGTH_INDEFINITE
        )
        val vjuz = snack!!.view
        vjuz.setBackgroundColor(ContextCompat.getColor(requireActivity(), R.color.red_nice))
        snack!!.setAction("PONOVI") {
            snack!!.dismiss()
            startFetching()
        }
        snack!!.setActionTextColor(ContextCompat.getColor(requireContext(), R.color.white))

        snack!!.show()
    }

    fun showSnackError() {
        snack = Snackbar.make(
            requireActivity().findViewById(R.id.coordinatorLayout), "Došlo je do " +
                    "pogreške", Snackbar.LENGTH_LONG
        )
        val okvir = snack!!.view
        okvir.setBackgroundColor(ContextCompat.getColor(requireActivity(), R.color.red_nice))
        snack!!.setAction("PONOVI") {
            snack!!.dismiss()
            startFetching()
        }
        snack!!.setActionTextColor(ContextCompat.getColor(requireContext(), R.color.white))
        snack!!.show()
    }

    private fun hideRecyc() {
        binding!!.nestedAttend.visibility = View.INVISIBLE
        binding!!.progressAttend.visibility = View.VISIBLE
    }

    override fun onStop() {
        super.onStop()
        if (snack != null) {
            snack!!.dismiss()
        }
        if (okHttpClient != null) okHttpClient!!.dispatcher.cancelAll()
        if (cRealm != null) cRealm!!.close()
    }

    override fun onDestroy() {
        super.onDestroy()
        if (nRealm != null) nRealm!!.close()
    }
}